File: client/src/app/shared/models/order.ts
import { DeliveryMethod } from "./deliveryMethod"

export interface Order {
  id: number
  orderDate: string
  buyerEmail: string
  shippingAddress: ShippingAddress
  deliveryMethod: DeliveryMethod
  shippingPrice: number
  paymentSummary: PaymentSummary
  orderItems: OrderItem[]
  subtotal: number
  discount?: number
  status: string
  total: number
  paymentIntentId: string
}

export interface ShippingAddress {
  name: string
  line1: string
  line2?: string
  city: string
  state: string
  postalCode: string
  country: string
}

export interface PaymentSummary {
  last4: number
  brand: string
  expMonth: number
  expYear: number
}

export interface OrderItem {
  itemOrdered: {
    productId: number;
    productName: string;
    pictureUrl: string;
  };
  price: number;
  quantity: number;
  id: number;
}


export interface OrderToCreate {
  cartId: string;
  deliveryMethodId: number;
  shippingAddress: ShippingAddress;
  paymentSummary: PaymentSummary;
  discount?: number;
}

File: Core/Entities/OrderAggreate/Order.cs
using Core.Interfaces;

namespace Core.Entities.OrderAggreate
{
    public class Order : BaseEntity
    {
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        public required string BuyerEmail { get; set; }

        public ShippingAddress ShippingAddress { get; set; } = null!;

        public DeliveryMethod DeliveryMethod { get; set; } = null!;

        public PaymentSummary PaymentSummary { get; set; } = null!;

        public List<OrderItem> OrderItems { get; set; } = [];

        public decimal Subtotal { get; set; }

        public OrderStatus Status { get; set; } = OrderStatus.Pending;

        public required string PaymentIntentId { get; set; }

        public AppCoupon? Coupon { get; set; } // الكوبون المستخدم في الطلب (اختياري)
        public decimal Discount { get; set; } // قيمة الخصم الناتجة عن الكوبون
        public decimal Total { get; set; } // المبلغ النهائي بعد الخصم والشحن


        public decimal GetTotal()
        {
         return Subtotal + DeliveryMethod.Price - (Discount > 0 ? Discount : 0);
        }

    }
}